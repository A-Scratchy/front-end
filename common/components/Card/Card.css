.Card {
  align-items: center;
  background-color: #ffffff;
  box-shadow: 0 3px 6px 2px rgba(0, 0, 0, 0.12);
  display: flex;
  flex-flow: column nowrap;
  justify-content: space-between;
  margin: 1rem;
  min-height: 100px;
  min-width: 100px;
  padding: 1.5rem;
  text-align: left;
}

.header {
  border-bottom: 1px solid var(--gray);
}

.content {
}

.footer {
  border-top: 1px solid var(--gray);
}
