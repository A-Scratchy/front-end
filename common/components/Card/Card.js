import React from 'react';
import PropTypes from 'prop-types';
import styles from './Card.css';

Card.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
};

Card.defaultProps = {
  className: undefined,
};

function Card({
  children, className,
}) {
  return <article className={`${styles.Card} ${className}`}>{children}</article>;
}

export default Card;
