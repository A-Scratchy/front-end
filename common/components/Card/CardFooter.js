import React from 'react';
import PropTypes from 'prop-types';
import styles from './Card.css';

CardFooter.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string,
};

CardFooter.defaultProps = {
  className: undefined,
};

function CardFooter({
  children, className,
}) {
  return <footer className={`${styles.footer} ${className}`}>{children}</footer>;
}

export default CardFooter;
