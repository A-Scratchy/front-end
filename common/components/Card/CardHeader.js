import React from 'react';
import PropTypes from 'prop-types';

CardHeader.propTypes = {
  children: PropTypes.node.isRequired,
  className: PropTypes.string.isRequired,
};

function CardHeader({
  children, className,
}) {
  return <header className={className}>{children}</header>;
}

export default CardHeader;
