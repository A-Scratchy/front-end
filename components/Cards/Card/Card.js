import classNames from 'classnames';
import { getDataAttributes } from 'common/utils/prop-utils';
import { bool, node, string } from 'prop-types';
import React from 'react';
import styles from './Card.css';

Card.propTypes = {
  children: node.isRequired,
  className: string,
  hasAnimationOnHover: bool,
};

Card.defaultProps = {
  className: undefined,
  hasAnimationOnHover: false,
};

function Card({ children, className, hasAnimationOnHover, ...props }) {
  const customDataAttributes = getDataAttributes(props);

  return (
    <article
      className={classNames(styles.Card, className, { [styles.animatedCard]: hasAnimationOnHover })}
      {...customDataAttributes}
    >
      {children}
    </article>
  );
}

export default Card;
