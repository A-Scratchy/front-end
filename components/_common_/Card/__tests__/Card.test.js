/* eslint-env jest */
import React from 'react';
import createSnapshotTest from 'test-utils/createSnapshotTest';

import Card from '../Card';

describe('Card', () => {
  it('should render with required props', () => {
    createSnapshotTest(<Card>Test</Card>);
  });

  it('should render with all props assigned', () => {
    createSnapshotTest(
      <Card className="test-class" hasAnimationOnHover>
        Test
      </Card>,
    );
  });
});
