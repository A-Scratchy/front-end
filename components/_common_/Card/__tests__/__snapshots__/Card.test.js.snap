// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should render with all props assigned 1`] = `
<article
  className="Card test-class animatedCard"
>
  Test
</article>
`;

exports[`Card should render with required props 1`] = `
<article
  className="Card"
>
  Test
</article>
`;
